packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -15,146 +15,122 @@
-          summary: 'Enforce branch protection on your repo.',
-          text:
-            'To add required status checks to your repository, please follow instructions in this link: \nhttps://help.github.com/en/github/administering-a-repository/enabling-required-status-checks\n' +
-            '\nIn order to add applications to your repository that will run check runs, please follow instructions here: \nhttps://developer.github.com/apps/installing-github-apps/\n' +
-            '\nLastly, please make sure that your required status checks are the same as the ones listed in your config file if you created one.',
-        },
-      });
+handler.listPRs = async function listPRs(): Promise<WatchPR[]> {
+  const query = datastore.createQuery(TABLE).order('created');
+  const [prs] = await datastore.runQuery(query);
+  const result: WatchPR[] = [];
+  for (const pr of prs) {
+    const created = new Date(pr.created).getTime();
+    const now = new Date().getTime();
+    console.info(`keystore data`, pr[datastore.KEY]);
+    const url = pr[datastore.KEY].name;
+    let state = 'continue';
+    if (now - created > MAX_TEST_TIME) {
+      console.warn(`deleting stale PR ${url}`);
+      await handler.removePR(url);
+      state = 'stop';
     }
+    const watchPr: WatchPR = {
+      number: pr.number,
+      repo: pr.repo,
+      owner: pr.owner,
+      state: state as 'continue' | 'stop',
+      url,
+    };
+    result.push(watchPr);
+  }
+  return result;
+};
 
-    if (branchProtection) {
-      if (branchProtection.required_status_checks.contexts.length < 3) {
-        checkParams = context.repo({
-          head_sha: commits[commits.length - 1].sha,
-          name: 'merge-on-green-readiness',
-          conclusion: 'failure' as Conclusion,
-          output: {
-            title: 'You have less than 3 required status checks',
-            summary:
-              "You likely don't have all the required status checks you need, please make sure to add the appropriate ones.",
-            text:
-              'To add required status checks to your repository, please follow instructions in this link: \nhttps://help.github.com/en/github/administering-a-repository/enabling-required-status-checks\n' +
-              '\nIn order to add applications to your repository that will run check runs, please follow instructions here: \nhttps://developer.github.com/apps/installing-github-apps/\n' +
-              '\nLastly, please make sure that your required status checks are the same as the ones listed in your config file if you created one.',
-          },
-        });
-      }
+handler.removePR = async function removePR(url: string) {
+  const key = datastore.key([TABLE, url]);
+  await datastore.delete(key);
+};
 
-      const branchProtectionArray = branchProtection.required_status_checks
-        .contexts as string[];
-      const configProtectionArray = configProtection as string[];
+handler.addPR = async function addPR(wp: WatchPR, url: string) {
```

Comment:
TODO: we can likely split the database functionality into its own file and import these helper functions for use in the main bot event handling.

Reply:
Also added this as well.